src/GlobalsAttributeReader.php

```diff
@@ -0,0 +1,125 @@
+    }
+
+    private function parseGlobalAttributes(TestMethod $method): array
+    {
+        return \array_map(static function (array $attributes) {
+            return \array_reduce($attributes, static function ($carry, $attribute) {
+                [$name, $value] = \strpos($attribute, '=') ? \explode('=', $attribute, 2) : [$attribute, ''];
+                $carry[$name] = $value;
+
+                return $carry;
+            }, []);
+        }, $this->findSetVarAttributes($method));
+    }
+
+    private function findSetVarAttributes(TestMethod $method): array
+    {
+        $attributes = $this->parseTestMethodAttributes($method);
+
+        return \array_filter(
+            \array_merge_recursive(
+                ['env' => [], 'server' => [], 'putenv' => []],
+                $attributes['class'] ?? [],
+                $attributes['method'] ?? []
+            ),
+            static fn(string $attributeName) => \in_array($attributeName, ['env', 'server', 'putenv']),
+            ARRAY_FILTER_USE_KEY
+        );
+    }
+
+    private function findUnsetVarAttributes(TestMethod $method): array
+    {
+        $attributes = $this->parseTestMethodAttributes($method);
+
+        return \array_filter(
+            \array_merge_recursive(
+                ['unset-env' => [], 'unset-server' => [], 'unset-putenv' => []],
+                $attributes['class'] ?? [],
+                $attributes['method'] ?? []
+            ),
+            static fn(string $attributeName) => \in_array($attributeName, ['unset-env', 'unset-server', 'unset-putenv']),
+            ARRAY_FILTER_USE_KEY
+        );
+    }
+
+    private function parseTestMethodAttributes(TestMethod $method): array
+    {
+        $className = $method->className();
+        $methodName = $method->methodName();
+
+        $methodAttributes = null;
+
+        if ($methodName !== null) {
+            $methodAttributes = $this->collectGlobalsFromAttributes((new \ReflectionMethod($className, $methodName))->getAttributes());
+        }
+
+        return [
+            'method' => $methodAttributes,
+            'class' => $this->collectGlobalsFromAttributes((new \ReflectionClass($className))->getAttributes() )
+        ];
+    }
+
+    private function collectGlobalsFromAttributes(array $attributes): array
+    {
+        $globals = [];
+
+        foreach ($attributes as $attribute) {
+            if (!str_starts_with($attribute->getName(), 'Zalas\\PHPUnit\\Globals\\Attribute\\')) {
+                continue;
+            }
+
+            $attributeInstance = $attribute->newInstance();
+            $attributeShortName = strtolower((new \ReflectionClass($attribute->getName()))->getShortName());
+
+            if ($attributeInstance->unset) {
+                $globals['unset-'.$attributeShortName][] = $attributeInstance->name;
+            } else {
+                $globals[$attributeShortName][] = sprintf('%s=%s', $attributeInstance->name, $attributeInstance->value);
+            }
+        }
```

Comment:
I see you were inspired by the way annotations are parsed into arrays. That's fine, but that was mostly dictated by the way PHPUnit parsed annotations. Feel free to come up with your own implementation if you want to.

Maybe the code could be simplified if we returned the collection of Attribute objects instead of the array? They then could be [matched](https://www.php.net/match) and dealt with based on type.

It's just a suggestion you can ignore. We're all busy.